eclass/dotnet-pkg-utils.eclass

```diff
@@ -0,0 +1,606 @@
+	RDEPEND+=" virtual/dotnet-sdk:${DOTNET_COMPAT} "
+	BDEPEND+=" ${RDEPEND} "
+
+	# Special package "dev-dotnet/csharp-gentoodotnetinfo" used for information
+	# gathering, example for usage see the "dotnet-pkg-utils_info" function.
+	if [[ ${CATEGORY}/${PN} != dev-dotnet/csharp-gentoodotnetinfo ]] ; then
+		BDEPEND+=" dev-dotnet/csharp-gentoodotnetinfo "
+	fi
+
+	IUSE+=" debug "
+fi
+
+# Needed otherwise the binaries may break.
+RESTRICT+=" strip "
+
+# Everything is built by "dotnet".
+QA_PREBUILT=".*"
+
+# Special .NET SDK environment variables.
+# Setting them either prevents annoying information from being generated
+# or stops services that may interfere with a clean package build.
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+export DOTNET_NOLOGO=1
+export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+export MSBUILDDISABLENODEREUSE=1
+export POWERSHELL_TELEMETRY_OPTOUT=1
+export POWERSHELL_UPDATECHECK=0
+# Overwrite selected MSBuild properties ("-p:XYZ").
+export UseSharedCompilation=false
+
+# @ECLASS_VARIABLE: DOTNET_RUNTIME
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Sets the runtime used to build a package.
+#
+# This variable is set automatically by the "dotnet-pkg-utils_setup" function.
+
+# @ECLASS_VARIABLE: DOTNET_EXECUTABLE
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Sets path of a "dotnet" executable.
+#
+# This variable is set automatically by the "dotnet-pkg-utils_setup" function.
+
+# @ECLASS_VARIABLE: DOTNET_CONFIGURATION
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Configuration value passed to "dotnet" in the compile phase.
+# Is either Debug or Release, depending on the "debug" USE flag.
+#
+# This variable is set automatically by the "dotnet-pkg-utils_setup" function.
+
+# @ECLASS_VARIABLE: DOTNET_OUTPUT
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Path of the output directory, where the package artifacts are placed during
+# the building of packages with "dotnet-pkg-utils_build" function.
+#
+# This variable is set automatically by the "dotnet-pkg-utils_setup" function.
+
+# @VARIABLE: DOTNET_LAUNCHERDEST
+# @INTERNAL
+# @DESCRIPTION:
+# Sets the path that .NET launchers are installed into by
+# the "dotnet-pkg-utils_dolauncher" function.
+#
+# The function "dotnet-pkg-utils_launcherinto" is able to manipulate this
+# variable.
+#
+# Defaults to "/usr/bin".
+DOTNET_LAUNCHERDEST=/usr/bin
+
+# @VARIABLE: DOTNET_LAUNCHERVARS
+# @INTERNAL
+# @DESCRIPTION:
+# Sets additional variables for .NET launchers created by
+# the "dotnet-pkg-utils_dolauncher" function.
+#
+# The function "dotnet-pkg-utils_append_launchervar" is able to manipulate this
+# variable.
+#
+# Defaults to a empty array.
+DOTNET_LAUNCHERVARS=()
+
+# @FUNCTION: dotnet-pkg-utils_get-configuration
+# @DESCRIPTION:
+# Return .NET configuration type of the current package.
+#
+# It is advised to refer to the "DOTNET_CONFIGURATION" variable instead of
+# calling this function if necessary.
+#
+# Used by "dotnet-pkg-utils_setup".
+dotnet-pkg-utils_get-configuration() {
+	if in_iuse debug && use debug ; then
```

Comment:
Do you actually need `in_iuse debug` here? FWICS it's only not there in `dotnet-runtime-nugets`, and I don't think you're calling this function there.

Reply:
> Do you actually need `in_iuse debug` here? FWICS it's only not there in `dotnet-runtime-nugets`, and I don't think you're calling this function there.

I think my point was to make debug able to be toggled included or not but I forgot about it.